File: LiverAblation/shared/scripts/ScreenObject.js
source(findFile("scripts","thing.js"));

   
var baseScreen = {   
    
    doSomething: function(){
        mouseClick("{column='1' container=':Form.customTreeWidget_CustomTreeWidget' text='1.3.6.1.4.1.9328.50.3.0537' type='QModelIndex'}");
        return startScreen;
    },
    doSomethingElse: function(){
        mouseClick("{container=':customTreeWidget.frame_QFrame_2' name='openPlanButton' text='Open Plan' type='QPushButton' visible='1'}");
        return startScreen;
    },
    enterText: function(thing, someText){
        mouseClick(waitForObject(thing));
        type(waitForObject(thing), someText);
        return startScreen;
    },
    grabScreen: function(thing){
        widget = waitForObject(thing);
        img = grabWidget(widget);
        img.save("C:\\Users\\SethUrban\\suite_LiverAblation_POC_2\\tst_case1\\kermit_screenshot.png", "PNG");
    }
}



var startScreen = {
    base:  baseScreen,
    textbox: new thing(":Form.search_QLineEdit", ":Form.search_QLineEdit"),
    
    searchfFor: function(sometext){
        this.base.enterText(this.textbox._location, sometext);
        this.base.grabScreen(this.textbox._location);
    }
}

File: LiverAblation/shared/scripts/ScreenObject.rb
# encoding: UTF-8
require findFile("scripts", "TestLogger.rb")

class BaseScreenObject
  def initialize
      @logFile = ""
    end
  
    def initLog(testName)
      @logFile = testName
    end
  
    def enterText(element, someText)
      mouseClick(waitForObject(element.symbolicName))
      @logFile.AppendLog("MouseClick on: " + element.symbolicName)
      type(waitForObject(element.symbolicName), someText)
      @logFile.AppendLog("Type " + someText + " into: " + element.symbolicName)
    end
  
end